core/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -470,6 +475,46 @@ public BigArrays getBigArrays() {
         return searchOperationListeners;
     }
 
+    @Override
+    public boolean requiresIndexMappingRefresh(IndexMetaData indexMetaData) throws IOException {
+        MapperService mapperService = mapperService();
```

Comment:
should this entire method go into `MapperService`? I think this would make more sense here